include/fi.h

```diff
@@ -144,38 +145,69 @@ static inline uint64_t roundup_power_of_two(uint64_t n)
 #endif /* PT_LOCK_SPIN */
 
 
+#if ENABLE_DEBUG
+#define ATOMIC_IS_INITIALIZED(atomic) assert(atomic->is_initialized)
+#else
+#define ATOMIC_IS_INITIALIZED(atomic)
+#endif
+
 #ifdef HAVE_ATOMICS
-typedef atomic_int atomic_t;
+typedef struct {
+    atomic_int val;
+#if ENABLE_DEBUG
+    int is_initialized;
+#endif
+} atomic_t;
 
 static inline int atomic_inc(atomic_t *atomic)
 {
-	return atomic_fetch_add_explicit(atomic, 1, memory_order_acq_rel) + 1;
+	ATOMIC_IS_INITIALIZED(atomic);
+	return atomic_fetch_add_explicit(&atomic->val, 1, memory_order_acq_rel) + 1;
 }
 
 static inline int atomic_dec(atomic_t *atomic)
 {
-	return atomic_fetch_sub_explicit(atomic, 1, memory_order_acq_rel) - 1;
+	ATOMIC_IS_INITIALIZED(atomic);
+	return atomic_fetch_sub_explicit(&atomic->val, 1, memory_order_acq_rel) - 1;
 }
 
 static inline int atomic_set(atomic_t *atomic, int value)
 {
-	atomic_store(atomic, value);
+	ATOMIC_IS_INITIALIZED(atomic);
+	atomic_store(&atomic->val, value);
 	return value;
 }
 
 static inline int atomic_get(atomic_t *atomic)
 {
-	return atomic_load(atomic);
+	ATOMIC_IS_INITIALIZED(atomic);
+	return atomic_load(&atomic->val);
+}
+
+/* avoid using "atomic_init" so we don't conflict with symbol/macro from stdatomic.h */
+static inline void atomic_initialize(atomic_t *atomic, int value)
+{
+	atomic_store(&atomic->val, value);
+#if ENABLE_DEBUG
+	atomic->is_initialized = 1;
+#endif
```

Comment:
No specific reason.  I didn't originally avoid using the `atomic_init` name when writing this patch because I didn't realize at first that it was part of stdatomic.h.